server/store/peer_store.go

```diff
@@ -24,6 +24,12 @@ type PeerStore interface {
 	GraduateLeecher(infoHash chihaya.InfoHash, p chihaya.Peer) error
 	AnnouncePeers(infoHash chihaya.InfoHash, seeder bool, numWant int) (peers, peers6 []chihaya.Peer, err error)
 	CollectGarbage(cutoff time.Time) error
+
+	GetSeeders(infoHash chihaya.InfoHash) (peers, peers6 []chihaya.Peer, err error)
+	GetLeechers(infoHash chihaya.InfoHash) (peers, peers6 []chihaya.Peer, err error)
```

Comment:
Sure, but they should also be able to reuse this store interface and just have an alternative memory implementation.